hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -652,8 +670,24 @@ private double ratioToBytes(Long nodeCapacity, double utilizationRatio) {
    */
   boolean canSizeEnterTarget(DatanodeDetails target, long size) {
     if (sizeEnteringNode.containsKey(target)) {
-      return sizeEnteringNode.get(target) + size <=
-          config.getMaxSizeEnteringTarget();
+      long sizeEnteringAfterMove = sizeEnteringNode.get(target) + size;
+      SCMNodeMetric scmNM = nodeManager.getNodeStat(target);
+      Preconditions.checkNotNull(scmNM);
+      SCMNodeStat scmNodeStat = scmNM.get();
+      double capacity = scmNodeStat.getCapacity().get();
+      Preconditions.checkArgument(capacity > 0);
+
+      //size can be moved into target datanode only when the following
+      //two condition are met.
+      //1 sizeEnteringAfterMove does not succeed the configured
+      // MaxSizeEnteringTarget
+      //2 current usage of target datanode plus sizeEnteringAfterMove
+      // is smaller than or equal to upperLimit
+      double usedAfterMove =
+          capacity - scmNodeStat.getRemaining().get() + sizeEnteringAfterMove;
+
```

Comment:
Can we move this calculation in a separate Utils class or sth like that? @siddhantsangwan had added a function in DatanodeUsageInfo I think?
I feel we can reuse the same function. We will only need a parameter for used size to add.

Reply:
`DatanodeUsageInfo.calculateUtilization`

Reply:
yea , i have updated this patch , PTAL!